source/level/TestGenState.hx

```diff
@@ -61,12 +72,25 @@ class TestGenState extends FlxState {
 		}
 	}
 
-	private function checkFinished() {
-		if (enemyTanks.countLiving() <= 0) {
-			FlxG.switchState(new FinishedLevelState());
-		}
-		else if (!playerTank.alive) {
-			FlxG.switchState(new StartMenuState());
+	private function doBulletCollision(bullets:FlxGroup) {
+		FlxG.collide(bullets, map, function onOverlap(bullets, map) {
+			bullets.impact("wall");
+		});
+		FlxG.overlap(bullets, enemyTanks, function onOverlap(bullets, enemyTank) {
+			bullets.impact("tank");
+		});
+		FlxG.overlap(bullets, playerTank, function onOverlap(bullets, playerTank) {
+			bullets.impact("tank");
+		});
+		FlxG.overlap(bullets, null, function onOverlap(bullets, na) {
+			bullets.impact("projectile");
+		});
+		private function checkFinished() {
+			if (enemyTanks.countLiving() <= 0) {
+				FlxG.switchState(new FinishedLevelState());
+			}
+			else if (!playerTank.alive) {
+				FlxG.switchState(new StartMenuState());
+			}
 		}
```

Comment:
This method shouldn't be defined inside of  `doBulletCollision`. I'm guessing a bracket got misplaced and jammed this into the wrong spot.

Reply:
it turns out we're also missing a `}` to close the class definition too.